api/app/utils/emailer.py

```diff
@@ -8,6 +8,54 @@
 from email.mime.base import MIMEBase
 from email import encoders
 from flask import g, request
+from app.email_template.repository import EmailRepository as email_repository
+from app.users.repository import UserRepository as user_repository
+from app.events.repository import EventRepository as event_repository
+
+def email_user(
+    email_template_key, 
+    template_parameters=None, 
+    event_id=None,
+    event=None,
+    user_id=None, 
+    user=None, 
+    subject_parameters=None, 
+    file_name='',
+    file_path=''
+):
+    """Send an email to a specified user using an email template. Handles resolving the correct language."""
+    if user_id is None and user is None:
+        raise ValueError('You must specify one of user_id or user')
+
+    user = user or user_repository.get_by_id(user_id)
+    if event is None and event_id is not None:
+        event = event_repository.get_by_id(event_id)
+    language = user.user_primaryLanguage
+    event_id = None if event is None else event.id
+    email_template = email_repository.get(event_id, email_template_key, language)
+
+    if email_template is None:
+        raise ValueError('Could not find email template with key {}'.format(email_template_key))
+    
+    subject_parameters = subject_parameters or {}
+    if event is not None and 'event_name' not in subject_parameters:
+        subject_parameters['event_name'] = event.get_name(language) if event.has_specific_translation(language) else event.get_name('en')
+
+    subject = email_template.subject.format(**subject_parameters)
+
+    template_parameters = template_parameters or {}
+    if 'title' not in template_parameters:
+        template_parameters['title'] = user.user_title
+    if 'firstname' not in template_parameters:
+        template_parameters['firstname'] = user.firstname
+    if 'lastname' not in template_parameters:
+        template_parameters['lastname'] = user.lastname
+    if event is not None and 'event_name' not in template_parameters:
+        template_parameters['event_name'] = event.get_name(language) if event.has_specific_translation(language) else event.get_name('en')
+
+    body_text = email_template.template.format(**template_parameters)
+    send_mail(recipient=user.email, subject=subject, body_text=body_text)
```

Comment:
I think you need to pass the `file_name` and `file_path` parameters here

Reply:
Good catch Thanks!!